src/main/java/com/github/martynagil/drugstoremanagement/service/EmployeeService.java

```diff
@@ -35,9 +35,11 @@ public void addNewEmployee(EmployeeDto employeeDto) {
     public void dismissEmployee(Long employeeId, EmployeeDismissalDto employeeDismissalDto) {
         Employee employee = employeeRepository.findById(employeeId)
                 .orElseThrow(() -> new EntityNotFoundException());
-        if (employee.getDateOfDismissal().equals(null)) {
+        if (employee.getDateOfDismissal() == null) {
             employee.dismiss(employeeDismissalDto.getDismissalDate());
             employeeRepository.save(employee);
+        } else {
+            throw new IllegalStateException();
```

Comment:
warunek powinien być odwrócony. Jeśli tak zrobisz utworzysz tak zwany "guard clause":
https://refactoring.guru/pl/replace-nested-conditional-with-guard-clauses
https://medium.com/@scadge/if-statements-design-guard-clauses-might-be-all-you-need-67219a1a981a

wtedy też nie jest w ogóle potrzebny else, co sprawia że kod jest czytelniejszy